### DISCUSSION TITLE
Is it possible to disable font antialiasing in WezTerm on macOS?

### DISCUSSION THREAD

**Discussion by bio** [wezterm/wezterm#2231] [Q&A]
I tried to find an option to disable font anti-aliasing but with no luck. Is it possible to disable the font anti-aliasing?

WezTerm and iTerm2 screenshots below.
[antialiasing.zip](https://github.com/wez/wezterm/files/9068235/antialiasing.zip)

WezTerm params:
```
return {
  font = wezterm.font_with_fallback({
    "Monaco",
  }),
  font_size = 16.0,
  freetype_load_target = "Normal",
  line_height = 1.02,
  cell_width = 1.04,
}
```

Thanks for the great terminal emulator!

**Comment by wez** [comment]
Freetype renders in 8-bit grayscale with anti-aliasing by default.
You can force it into 1-bit color by setting `freetype_load_flags = "MONOCHROME"` but you'll probably hate how it looks.

Why do you want to disable anti-aliasing?

The screenshot you shared looks like there might be a hinting issue; hinting in fonts is a process where the rasterizer tries to align fonts to the "middle" of a pixel rather than the "edge" of a pixel to have better definition for lines.  That can sometimes be at-odds with opengl programs.

You may wish to try setting `freetype_load_flags = "NO_HINTING"` to disable that and see if it improves things for you.

**Comment by bio** [comment]
@wez Thank you for the answer, I tried to set MONOCHROME/NO_HINTING/etc freetype_load_flags flags but still see the grey pixels around the letters.

Eyes are less tired if the transitions between colors are not smooth. Then it's easier for the eye to recognize the character.
That's why I prefer aliased text (especially on retina displays).

https://ask.libreoffice.org/t/the-price-of-smooth-looks-anti-aliased-fonts-hurt-eyes-and-damage-eyesight/24560/4
https://annystudio.com/try/sharp-text/

I'm used to MacVim font rendering with the `noantialias` option
```
set guifont=Monaco:h16
set noantialias
```
Also the text looks thinner than the text with anti-aliasing.

Screenshot: [macvim.png.zip](https://github.com/wez/wezterm/files/9073364/macvim.png.zip)

**Reply by wez** [reply]
You need to set both of these to prevent freetype from using anti-aliasing:

```
freetype_load_flags="MONOCHROME",
freetype_load_target="Mono",
```

**Reply by bio** [reply]
Awesome, thank you!
